source/language/openpulse.rst

```diff
@@ -0,0 +1,609 @@
+
+  float[32] fq_01 = 5e9; // hardcode or pull from some function
+  float[32] anharm = 300e6; // hardcode or pull from some function
+  cal {
+      frame frame_01 = newframe(fq_01, 0);
+      frame frame_12 = newframe(fq_01 + anharm, 0);
+  }
+
+  defcal geo_gate(angle[32] theta) $0 {
+      // theta: rotation angle (about z-axis) on Bloch sphere
+
+      // Access globally defined channels
+      tx_channel dq = txch($q, “drive”);
+
+      // Assume we have calibrated 0->1 pi pulses and 1->2 pi pulse
+      // envelopes (no sideband)
+      waveform X_01 = {...};
+      waveform X_12 = {...};
+      float[32] a = sin(theta/2);
+      float[32] b = sqrt(1-a**2);
+
+      // Double-tap
+      play(dq, scale(a, X_01), frame_01);
+      play(dq, scale(b, X_12), frame_12);
+      play(dq, scale(a, X_01), frame_01);
+      play(dq, scale(b, X_12), frame_12);
+  }
+
+Neutral atoms
+~~~~~~~~~~~~~
+
+In this simple example, the signal chain is composed of two electro-optic modulators (EOM) and
+an acousto-optic deflector (AOD). The EOMs put sidebands on the laser light while the AOD diffracts
+the light in an amount proportional to the frequency of the RF drive. This example was chosen
+because it is similar in spirit to the work by Levine et al.:cite:`levine2019` except that phase
+control is exerted using virtual Z gates on the AODs -- requiring frame tracking of the qubit
+frequency yet application of a tone that maps to the qubit position (i.e. requires the use of a
+sideband).
+
+The program aims to perform a Hahn echo sequence on q1, and a Ramsey sequence on q2 and q3.
+
+.. code-block:: javascript
+
+  defcal neutral_atoms {
+    // Access globally defined channels
+    channel eom_a_channel = txch(0, "eom_a");
+    channel eom_a_channel = txch(1, "eom_b");
+    channel aod_channel = txch(0, "aod");
+
+    // Define the Raman frames, which are detuned by an amount Δ from the  5S1/2 to 5P1/2 transition
+    // and offset from each other by the qubit_freq
+    frame raman_a_frame = newframe(Δ, 0.0)
+    frame raman_b_frame = newframe(Δ-qubit_freq, 0.0)
+
+    // Three copies of qubit freq to track phase of each qubit
+    frame q1_frame = newframe(qubit_freq, 0)
+    frame q2_frame = newframe(qubit_freq, 0)
+    frame q3_frame = newframe(qubit_freq, 0)
+
+    // Generic gaussian envelope
+    waveform π_half_sig = gaussian(..., π_half_time, ...)
+
+    // Waveforms ultimately supplied to the AODs. We mix our general Gaussian pulse with a sine wave to
+    // put a sideband on the signal construction to target the qubit position while maintainig the
+    // desired Rabi rate.
+    waveform q1_π_half_sig = mix(π_half_sig, sine(q1_π_half_amp, q1_pos_freq-qubit_freq, 0.0, π_half_time));
+    waveform q2_π_half_sig = mix(π_half_sig, sine(q2_π_half_amp, q2_pos_freq-qubit_freq, 0.0, π_half_time));
+    waveform q3_π_half_sig = mix(π_half_sig, sine(q3_π_half_amp, q3_pos_freq-qubit_freq, 0.0, π_half_time));
+
+    for τ in [0: T]:
```

Comment:
`T` not defined--maybe add to `cal` block?

Reply:
Also the `qubit_freq, Δ, tppi_1, tppi_2,` and `tppi_3`.

Reply:
I left a warning comment in the example about this in ea42e73.